aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -360,34 +368,54 @@ DimVector _estimate_sparse_compressed_tensor_size(
     const Tensor& plain_indices,
     const Tensor& values,
     Layout layout) {
-  int block_ndim = AT_DISPATCH_PLAIN_SPARSE_COMPRESSED_LAYOUTS(layout, "estimate_sparse_compressed_tensor_size", [&] { return 0; }, [&] { return 2; });
-  int dense_ndim = values.dim() - compressed_indices.dim() - block_ndim;
+  const int block_ndim = AT_DISPATCH_PLAIN_SPARSE_COMPRESSED_LAYOUTS(layout, "estimate_sparse_compressed_tensor_size", [&] { return 0; }, [&] { return 2; });
+  const int base_ndim = 2;  // corresponds to compressed and plain indices
+  const int batch_ndim = compressed_indices.dim() - 1;
+  const std::string compressed_indices_name = compressedIndicesName(layout);
+  const std::string plain_indices_name = plainIndicesName(layout);
+  TORCH_CHECK(
+              batch_ndim >= 0,
+              compressed_indices_name, " must have dimensionality >= 1 but got ", compressed_indices.dim());
+  TORCH_CHECK(
+              compressed_indices.dim() == plain_indices.dim(),
+              compressed_indices_name, " and ", plain_indices_name, " dimensionalities must be equal but got ",
```

Comment:
From here and below: don't we have all these checked in `_validate_sparse_compressed_tensor_args_worker`? Could this method be called before `_validate_sparse_compressed_tensor_args_worker`?

Reply:
Yes, these checks are run twice when a user does not specify the tensor size: in `_estimate_sparse_compressed_tensor_size` followed by `_validate_sparse_compressed_tensor_args_worker` call. However, we'll need these checks here to make sure that the inputs to `_estimate_sparse_compressed_tensor_size` are valid similar to `_validate_sparse_compressed_tensor_args_worker` that assumes a specified size. Notice that the overhead of these checks is minimal (it is about comparing two integers), so, performance-wise, it should be ok.

No, `_estimate_sparse_compressed_tensor_size` cannot be called before `_validate_sparse_compressed_tensor_args_worker` in general because (i) when an user specifies the `size`, there is no need to estimate the size from inputs, and (ii) the sizes returned by `_estimate_sparse_compressed_tensor_size` and specified by the user may be different while both being valid sizes (`_estimate_sparse_compressed_tensor_size` returns a *minimal* valid size for the given input).